### PR TITLE
api(ticdc): add changefeed unit tests

### PR DESCRIPTION
**Pull request by crelax**
### What problem does this PR solve?

Issue Number: close #xxx

### What is changed and how it works?
add unit tests of CDC Open API v2, and uplift the ut coverage to 73%

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
`None`
```

### CODE REVIEW

**Comment by crelax** [comment]
This PR was previously once proposed but closed: https://github.com/pingcap/tiflow/pull/6117

**Comment by crelax** [comment]
This PR was previously proposed but closed: https://github.com/pingcap/tiflow/pull/6117

**Review by asddongmen — APPROVED** [review]

File: cdc/api/v1/api.go [hunk 1]
```diff
@@ -263,7 +263,15 @@ func (h *OpenAPI) CreateChangefeed(c *gin.Context) {
 		return
 	}
 
-	up := h.capture.GetUpstreamManager().GetDefaultUpstream()
+	upManager, err := h.capture.GetUpstreamManager()
+	if err != nil {
+		_ = c.Error(err)
```
Comment (liuzix) [inline cdc/api/v1/api.go:268]:
```suggestion
		_ = c.Error(err)
		return
```

File: cdc/api/v1/api.go [hunk 2]
```diff
@@ -263,7 +263,15 @@ func (h *OpenAPI) CreateChangefeed(c *gin.Context) {
...
+	}
+	up, err := upManager.GetDefaultUpstream()
+	if err != nil {
+		_ = c.Error(err)
```
Comment (liuzix) [inline cdc/api/v1/api.go:272]:
```suggestion
		_ = c.Error(err)
		return
```

File: cdc/api/v2/api_helpers.go [hunk 3]
```diff
@@ -64,9 +83,328 @@ type APIV2Helpers interface {
 	// getPDClient returns a PDClient given the PD cluster addresses and a credential
 	getPDClient(context.Context, []string, *security.Credential) (pd.Client, error)
```
Comment (liuzix) [inline cdc/api/v2/api_helpers.go:84]: Please name arguments. For example, name the `[]string` as `endpoints`.

File: cdc/api/v2/model.go [hunk 4]
```diff
@@ -317,6 +315,19 @@ type RunningError struct {
 	Message string `json:"message"`
 }
 
+// toCredential generates a security.Credential from a PDConfig
+func (cfg *PDConfig) toCredential() *security.Credential {
```
Comment (liuzix) [inline cdc/api/v2/model.go:319]: Does `toCredential` have a unit test?

File: cdc/api/v2/model_test.go [hunk 5]
```diff
@@ -43,7 +43,6 @@ func TestToAPIReplicaConfig(t *testing.T) {
 			},
 		},
 		SchemaRegistry: "bbb",
-		TxnAtomicity:   "aa",
```
Comment (sdojjy) [inline cdc/api/v2/model_test.go:46]: why remove this line?

File: errors.toml [hunk 6]
```diff
@@ -1121,6 +1126,11 @@ error = '''
 updating service safepoint failed
 '''
 
+["CDC:ErrUpstreamManagerNotReady"]
+error = '''
+upstream manager not read
```
Comment (sdojjy) [inline errors.toml:1131]: not ready

File: pkg/errors/cdc_errors.go [hunk 7]
```diff
@@ -1020,6 +1024,11 @@ var (
 		errors.RFCCodeText("CDC:ErrUpstreamNotFound"),
 	)
 
+	ErrUpstreamManagerNotReady = errors.Normalize(
+		"upstream manager not read",
```
Comment (sdojjy) [inline pkg/errors/cdc_errors.go:1028]: ready

File: cdc/api/v2/model.go (hunk 4 above)
Comment (crelax) [reply cdc/api/v2/model.go:319]: added

**Review by sdojjy — APPROVED** [review]

**Pull request merged**